docs/design/data_model.md

```diff
@@ -0,0 +1,412 @@
+
+### Chain DHT Operations
+
+Represent chain data being shared with the network. Each action generates a specific set of chain operations (such as _StoreRecord_, _RegisterAgentActivity_, _StoreEntry_, etc.) that enable different authorities to store and index the data for efficient queries.
+
+### Warrant Operations
+
+Represent evidence of invalid behavior by an agent. When an agent detects that another agent has violated validation rules or forked their chain, they can issue a warrant containing cryptographic proof of the violation. Warrants are routed to the violating agent's activity authority, creating a permanent, queryable record of the misbehavior.
+
+### Location
+
+A location is a derived property of a hash. The exact algorithm is not important, as long as it preserves the uniqueness of its hash and all peers are using the same algorithm. The location value is an unsigned 32-bit number.
+
+## Actions taken by an agent
+
+### Action types
+
+There are 8 action types: _Dna_, _AgentValidationPkg_, _InitZomesComplete_, _Create_, _Update_, _Delete_, _CreateLink_, _DeleteLink_. The following sections describe their semantic meaning.
+
+#### Genesis
+
+When an agent first joins a DNA network, their source chain must be initialized through a process called _genesis_. Genesis creates the first three actions on every source chain.
+
+The genesis actions are:
+
+1. _Dna_ (sequence 0) - Records the DNA hash of the application this agent is joining.
+2. _AgentValidationPkg_ (sequence 1) - Contains the agent's public key and optional membrane proof for network validation.
+3. _Create_ (sequence 2) - Creates an _Agent_ typed entry, containing the agent's public key.
+
+These three genesis actions are created automatically by the system during the genesis workflow. The _Dna_ and _AgentValidationPkg_ actions cannot be created through application code, only queried. The _Create_ action with an _Agent_ typed entry can be created by an application but has special meaning in this location.
+
+The _Dna_ action declares what application data model and validation rules the agent is using. The _AgentValidationPkg_ and _Create_ actions "announce" the agent to
+the network and set up the expectation that all further actions on this chain will be signed by this agent.
+
+#### Initialisation
+
+After genesis completes successfully, an application is allowed to perform initialization steps. This may optionally choose to create actions of other types, to be described below. Once initialization is complete, an _InitZomesComplete_ action is appended to signal that the agent is ready to participate in the network.
+
+The _InitZomesComplete_ is required to guarantee that the application's initialisation steps are only run once. This is because, to Holochain, initialisation actions are
```

Comment:
```suggestion
The _InitZomesComplete_ is required to guarantee that the application's post-genesis initialisation Actions are committed only once. This is because, to Holochain, initialisation Actions are
```